include/groonga/groonga.h

```diff
@@ -965,14 +965,52 @@ grn_obj_get_values(grn_ctx *ctx, grn_obj *obj, grn_id offset, void **values);
 /**
  * \brief Subtract the specified value from the record/column value
  */
-#define GRN_OBJ_DECR    (0x03)
-#define GRN_OBJ_APPEND  (0x04)
+#define GRN_OBJ_DECR (0x03)
+/**
+ * \brief Append the specified value to the column value.
+ */
+#define GRN_OBJ_APPEND (0x04)
+/**
+ * \brief Prepend the specified value to the column value.
+ */
 #define GRN_OBJ_PREPEND (0x05)
-#define GRN_OBJ_GET     (0x01 << 4)
+/**
+ * \brief Set the value of the new record/cloumn to value.
+ */
+#define GRN_OBJ_GET (0x01 << 4)
+/**
+ * \brief Compare the record/column value with value.
+ */
 #define GRN_OBJ_COMPARE (0x01 << 5)
-#define GRN_OBJ_LOCK    (0x01 << 6)
-#define GRN_OBJ_UNLOCK  (0x01 << 7)
+/**
+ * \brief Lock the specified record/column value.
+ *        If used with \ref GRN_OBJ_COMPARE, the lock is applied only when the
+ *        record/column value matches value.
+ */
+#define GRN_OBJ_LOCK (0x01 << 6)
+/**
+ * \brief Unlock the specified record/column value.
+ */
+#define GRN_OBJ_UNLOCK (0x01 << 7)
 
+/**
+ * \brief Update the value of a record identified by the given ID in the
+ *        specified object.
+ *
+ *        This function updates the value of the record specified by `id` in the
+ *        given object (`obj`). If the corresponding record does not exist, it
+ *        returns \ref GRN_INVALID_ARGUMENT.
+ *
+ * \param ctx The context object.
+ * \param obj The target object where the value will be updated.
+ * \param id The ID of the record to be updated.
+ * \param value The value to be stored in the record.
+ * \param flags One of \ref GRN_OBJ_SET \ref GRN_OBJ_INCR \ref GRN_OBJ_DECR.
```

Comment:
We can use `GRN_OBJ_APPEND`/`GRN_OBJ_PREPEND` for vector column.

Reply:
> We can use GRN_OBJ_APPEND/GRN_OBJ_PREPEND for vector column.

Thanks. I misunderstood the previous comment. It is used like the following. 
- `grn_obj_set_value` -> `grn_obj_set_value_column_var_size` -> `grn_ja_putv`

Reply:
fix: [c94befa](https://github.com/groonga/groonga/pull/1902/commits/c94befae5d49fdcaa12896627403f77962e0b551)[](url)